### PR TITLE
Change validator set during end-to-end sync test

### LINKED ISSUE #1697: End-to-end sync tests should change the validator set
### Expected Behavior

We change the validator set in the end-to-end sync tests to get more coverage of ultralight sync + validator set changes.

### Current Behavior

Static validator set

### PR DESCRIPTION
**Pull request by asaj**
### Description

This PR updates the end-to-end sync test to use a changing validator set. This is important as it tests that ultralight sync properly applies the validator set diff when validating headers. 

### Tested

These are tests.

Fixes #1697

### CODE REVIEW

File: packages/celotool/src/e2e-tests/sync_tests.ts [hunk 1]
```diff
@@ -28,18 +36,54 @@ describe('sync tests', function(this: any) {
...
+    const validators = await (await validatorKit._web3Contracts.getValidators()).methods
+      .getRegisteredValidators()
+      .call()
+    const web3 = new Web3('ws://localhost:8557')
+    await sleep(10)
     await waitToFinishSyncing(web3)
+    const kit = newKitFromWeb3(web3)
+    // The validator set size at any one time will be 4, and we rotate two validators per epoch.
+    // This means that if a node were to have a static view of the validator set, we guarantee that
+    // the node will reject blocks.
```
Comment (jfoutts-celo) [inline packages/celotool/src/e2e-tests/sync_tests.ts]: Maybe add reference or link to the celo-blockchain code that will reject the block due to differing validator set?

File: packages/celotool/src/e2e-tests/sync_tests.ts [hunk 2]
```diff
@@ -28,18 +36,54 @@ describe('sync tests', function(this: any) {
...
+      rpcport: 8559,
+      privateKey: groupPrivateKey.slice(2),
       peers: [8545],
     }
     await initAndStartGeth(hooks.gethBinaryPath, fullInstance)
-    const web3 = new Web3('http://localhost:8553')
+    const validators = await (await validatorKit._web3Contracts.getValidators()).methods
+      .getRegisteredValidators()
+      .call()
+    const web3 = new Web3('ws://localhost:8557')
```
Comment (jfoutts-celo) [inline packages/celotool/src/e2e-tests/sync_tests.ts:65]: Nice!  We should probably favor ws over http regardless of needing `eth.subscribe()`.

**Review by jfoutts-celo — APPROVED** [review]

File: packages/celotool/src/e2e-tests/sync_tests.ts (hunk 2 above)
Comment (asaj) [reply packages/celotool/src/e2e-tests/sync_tests.ts:65]: Totally agree, I'll create tech debt issue for this

Comment (asaj) [reply packages/celotool/src/e2e-tests/sync_tests.ts:65]: https://github.com/celo-org/celo-monorepo/issues/2616

File: packages/celotool/src/e2e-tests/sync_tests.ts (hunk 1 above)
Comment (asaj) [reply packages/celotool/src/e2e-tests/sync_tests.ts]: Explicitly pointed out the function that would return false

**Pull request closed without merging**